bot/cogs/doc/cog.py

```diff
@@ -3,12 +3,13 @@
 import logging
```

Comment:
The functionality the changes here implement is that instead of caching beautifulsoup objects like in the current implementation https://github.com/python-discord/bot/pull/1014 (which can bubble up and potentially become a memory issue) we store the lookup ids of each symbol, and when one symbol on a page is requested all others on that page are parsed alongside it so we can throw out the BS object when that's done.